sound/soc/sof/sof-audio.c

```diff
@@ -508,6 +555,16 @@ void sof_machine_unregister(struct snd_sof_dev *sdev, void *pdata)
 {
 	struct snd_sof_pdata *plat_data = pdata;
 
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_NOCODEC)
+	if (!IS_ERR_OR_NULL(plat_data->nocodec_mach)) {
```

Comment:
can it be ERR? I only see an assignment in the success case line 535
plat_data->nocodec_mach = auxdev